h/auth.py

```diff
@@ -136,9 +137,14 @@ def groupfinder(userid, request):
     """
     principals = set()
 
-    user = models.User.get_by_userid(userid)
+    username, _ = util.split_user(userid)
+    if username is None:
```

Comment:
Once this implements all the same checks as `h.accounts.authenticated_user`, it's going to be a fair bit of duplication. Again, I'd suggest implementing the logic in a function which accepts a userid, not a request, and returns a user object or None, and put that in `h.accounts`. That function can then be used by this module, and by the function that you use as a request method:

##### `accounts/__init__.py`:
```python
def get_user(userid):
    ...

def includeme(config):
    config.add_request_method(lambda req: get_user(req.authenticated_userid), 
                                                   name='authenticated_user', 
                                                   reify=True)
```

##### `auth.py`:
```python
from h import accounts

def groupfinder(userid, request):
    ...
    user = accounts.get_user(userid)
    ...
```